tests/test_ssa_generation.py

```diff
@@ -1077,3 +1079,28 @@ def test_issue_1748():
         operations = f.slithir_operations
         assign_op = operations[0]
         assert isinstance(assign_op, InitArray)
+
+
+def test_issue_1776():
+    source = """
+    contract Contract {
+        function foo() public returns (uint) {
+            uint[5][10][] memory arr = new uint[5][10][](2);
+            return 0;
+        }
+    }
+    """
+    with slither_from_source(source) as slither:
+        c = slither.get_contract_from_name("Contract")[0]
+        f = c.functions[0]
+        operations = f.slithir_operations
+        new_op = operations[0]
+        lvalue = new_op.lvalue
+        lvalue_type = lvalue.type
+        assert isinstance(lvalue_type, ArrayType)
+        lvalue_type1 = lvalue_type.type
```

Comment:
```suggestion
        lvalue_type1 = lvalue_type.type
        assert lvalue_type1.is_dynamic
```
We can add the assertion that this array is dynamically sized and perhaps the inverse for those statically sized

Reply:
It seems to me that only lvalue_type should be dynamic.
lvalue_type1 and lvalue_type2 are statically sized.
I have updated the test according to your suggestion in 4768a08.